plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx

```diff
@@ -226,3 +228,53 @@ export const withUpgradeNoticeControls =
 	};
 
 addFilter( 'editor.BlockEdit', metadata.name, withUpgradeNoticeControls );
+
+export const withHandlePreviewState =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
+	( props: BlockEditProps< ProductCollectionAttributes > ) => {
+		if ( ! isProductCollection( props.name ) ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		let previewMessage = __(
+			'Actual products will vary depending on the currently viewed page.',
+			'woocommerce'
+		);
+
+		/**
+		 * Example:
+		 * How to change the preview message based on the collection selected.
+		 */
+		if ( props.attributes.collection === CoreCollectionNames.ON_SALE ) {
+			previewMessage = __(
+				'Custom tooltip for on sale collection.',
+				'woocommerce'
+			);
+		}
+
+		/**
+		 * 3PDs will be able to pass handlePreviewState as a prop & then update the state
+		 * based on their own logic.
+		 * Example: As you can see below, we are updating the state after 5 seconds.
+		 *          Also, we are toggling the isPreview state in each 5 seconds.
+		 */
+		const handlePreviewState = (
+			previewState: PreviewState,
+			setPreviewState: React.Dispatch<
+				React.SetStateAction< PreviewState >
+			>
+		) => {
+			setTimeout( () => {
+				setPreviewState( {
+					isPreview: ! previewState.isPreview,
+					previewMessage,
+				} );
+			}, 5000 );
+		};
+
+		return (
+			// Here we are passing the handlePreviewState as a prop to the BlockEdit.
+			<BlockEdit { ...props } handlePreviewState={ handlePreviewState } />
```

Comment:
Question: 

do you really need to have a function passed in to change the state? React has props already for this purpose.

If you want to update the component every 5 seconds then you just handle the state here in the wrapper. I think thats less complicated than requiring users to pass a function to do the state update.

Reply:
That's a good question, and I appreciate you bringing it up! In light of your query, I've made a few adjustments to this PR. As outlined in the PR description, we introduced a function called `registerProductCollection` which is pivotal for every collection to seamlessly register itself. While it's currently for internal use, our vision is to eventually open it up for 3PDs to utilize. This function is designed to be compatible with all parameters supported by [Block Variation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/#defining-a-block-variation).

Following your question, and as detailed in the [recent updates](https://github.com/woocommerce/woocommerce/blob/7950dfd5a9a6c397ddb3b41fc79e6236e7f2948c/plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/register-product-collection.tsx#L26-L59), our aim is to avoid necessitating collections to construct a wrapper around 'editor.BlockEdit'. This responsibility is efficiently handled by the registerProductCollection function, thereby necessitating the need for a callback function to manage such integrations.

I hope this clarifies the rationale behind these changes! 🙂